src/Field.test.js

```diff
@@ -706,4 +706,68 @@ describe('Field', () => {
     expect(barValidate).toHaveBeenCalledTimes(1)
     expect(bazValidate).toHaveBeenCalledTimes(1)
   })
+
+  it('should warn when used without type prop and rendering radio, checkbox or multiple select indirectly', () => {
+    class Container extends React.Component {
+      render() {
+        return (
+          <Form initialValues={{ select_test: [] }} onSubmit={() => {}}>
+            {() => (
+              <div>
+                <Field name="checkbox_test" value="checkbox_value">
+                  {({ input }) => <input type="checkbox" {...input} />}
+                </Field>
+                <Field name="radio_test" value="radio_value">
+                  {({ input }) => <input type="radio" {...input} />}
+                </Field>
+                <Field name="select_test">
+                  {({ input }) => (
+                    <select multiple {...input}>
+                      <option>{'Option'}</option>
+                    </select>
+                  )}
+                </Field>
+              </div>
+            )}
+          </Form>
+        )
+      }
+    }
+
+    const spy = jest.spyOn(global.console, 'error').mockImplementation(() => {})
+    const dom = TestUtils.renderIntoDocument(<Container />)
+
+    const [checkbox, radio] = TestUtils.scryRenderedDOMComponentsWithTag(
+      dom,
+      'input'
+    )
+    const [select] = TestUtils.scryRenderedDOMComponentsWithTag(dom, 'select')
+
+    TestUtils.Simulate.change(checkbox, {
+      target: { type: 'checkbox', value: 'checkbox_value' }
+    })
+    TestUtils.Simulate.change(radio, {
+      target: { type: 'radio', value: 'radio_value' }
+    })
+    TestUtils.Simulate.change(select, {
+      target: { type: 'select-multiple', value: ['select_value'] }
+    })
+
+    expect(spy).toHaveBeenCalled()
+    expect(spy).toHaveBeenCalledTimes(3)
+    expect(spy.mock.calls[0][0]).toBe(
```

Comment:
I find snapshots really nice for those kind of tests ("warning" text/errors thrown snapshots), but you do not use any snapshots so I wasn't sure if you'd like this approach.

Reply:
I only just recently understood snapshot testing. Not actively against it, but not excited to migrate any tests that could be more concise with snapshots... 🤷‍♂️